### PR TITLE
Feature historian

### PR DESCRIPTION
**Pull request by mbwhite**
## Checklist
 - [ ]  A link to the issue/user story that the pull request relates to
 - [ ]  How to recreate the problem without the fix
 - [ ]  Design of the fix
 - [ ]  How to prove that the fix works
 - [ ]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Steps to Reproduce

1.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

### CODE REVIEW

File: packages/composer-client/changelog.txt [hunk 1]
```diff
@@ -12,6 +12,9 @@
 # Note that the latest public API is documented using JSDocs and is available in api.txt.
 #
 
+Version 0.9.4 {f73a45b584fadd644734a91fe4cbbadc} 2017-07-31
```
Comment (davidkel) [inline packages/composer-client/changelog.txt]: Should it be Version 0.10.2 ?

File: packages/composer-client/lib/registry.js [hunk 2]
```diff
@@ -173,10 +176,13 @@ class Registry {
         if (!resources) {
             throw new Error('resources not specified');
         }
-        let serializedResources = resources.map((resource) => {
-            return this.serializer.toJSON(resource);
-        });
-        return Util.invokeChainCode(this.securityContext, 'addAllResourcesToRegistry', [this.registryType, this.id, JSON.stringify(serializedResources)]);
+        let txName = 'Add'+this.registryType;
+        const transaction = this.factory.newTransaction('org.hyperledger.composer.system',txName);
+        // transaction.targetRegistry = this.factory.newRelationship(ModelUtil.getNamespace(this.id),this.registryType, this.id);
```
Comment (davidkel) [inline packages/composer-client/lib/registry.js]: To be removed ? Or something to come in the future ? If so a comment would help here.

File: packages/composer-client/lib/registry.js [hunk 3]
```diff
@@ -207,10 +212,17 @@ class Registry {
...
-        });
-        return Util.invokeChainCode(this.securityContext, 'updateAllResourcesInRegistry', [this.registryType, this.id, JSON.stringify(serializedResources)]);
+        let txName = 'Update'+this.registryType;
+        const transaction = this.factory.newTransaction('org.hyperledger.composer.system',txName);
+        // transaction.targetRegistry = this.factory.newRelationship('org.hyperledger.composer.system',TYPE_MAP[this.registryType], this.id);
+        transaction.resources = resources;
+        transaction.registryType = this.registryType;
+        transaction.registryId = this.id;
+        return this.bnc.submitTransaction(transaction);
+        // let serializedResources = resources.map((resource) => {
```
Comment (davidkel) [inline packages/composer-client/lib/registry.js]: Code to keep for later or to remove ?

File: packages/composer-client/lib/registry.js [hunk 4]
```diff
@@ -235,20 +246,29 @@ class Registry {
      * @param {(Resource[]|string[])} resources The resources, or the unique identifiers of the resources.
      * @return {Promise} A promise that will be resolved when the resource is
      * added to the registry.
+     * TODO: need to change this
```
Comment (davidkel) [inline packages/composer-client/lib/registry.js]: change this to ?

File: packages/composer-client/lib/registry.js [hunk 5]
```diff
@@ -235,20 +246,29 @@ class Registry {
...
      */
     removeAll(resources) {
         Util.securityCheck(this.securityContext);
         if (!resources) {
             throw new Error('resources not specified');
         }
-        let data = resources.map((resource) => {
+        let txName = 'Remove'+this.registryType;
+        const transaction = this.factory.newTransaction('org.hyperledger.composer.system',txName);
+        // transaction.targetRegistry = this.factory.newRelationship('org.hyperledger.composer.system',TYPE_MAP[this.registryType], this.id);
```
Comment (davidkel) [inline packages/composer-client/lib/registry.js]: Reason to keep or remove ?

File: packages/composer-client/lib/registry.js [hunk 6]
```diff
@@ -235,20 +246,29 @@ class Registry {
...
+        transaction.resourceIds = resources.map((resource) => {
             if (resource instanceof Resource) {
                 return resource.getIdentifier();
             } else {
                 return resource;
             }
         });
-        return Util.invokeChainCode(this.securityContext, 'removeAllResourcesFromRegistry', [this.registryType, this.id, JSON.stringify(data)]);
+        return this.bnc.submitTransaction(transaction);
+        // return Util.invokeChainCode(this.securityContext, 'removeAllResourcesFromRegistry', [this.registryType, this.id, JSON.stringify(data)]);
```
Comment (davidkel) [inline packages/composer-client/lib/registry.js]: remove ?

File: packages/composer-client/test/registry.js [hunk 7]
```diff
@@ -20,6 +20,8 @@ const Registry = require('../lib/registry');
 const Resource = require('composer-common').Resource;
 const SecurityContext = require('composer-common').SecurityContext;
 const Serializer = require('composer-common').Serializer;
+const TransactionDeclartion = require('composer-common').TransactionDeclaration;
```
Comment (davidkel) [inline packages/composer-client/test/registry.js]: spelling....

File: packages/composer-common/lib/introspect/relationshipdeclaration.js [hunk 8]
```diff
@@ -77,7 +77,7 @@ class RelationshipDeclaration extends Property {
                 throw new IllegalModelException('Relationship ' + this.getName() + ' points to a missing type ' + this.getFullyQualifiedTypeName(), classDecl.getModelFile(), this.ast.location);
             }
 
-            if ((namespace === ModelUtil.getSystemNamespace()) && classDecl.isEvent()) {
+            if ( namespace === ModelUtil.getSystemNamespace() ) /*&& classDecl.isEvent())*/ {
```
Comment (davidkel) [inline packages/composer-common/lib/introspect/relationshipdeclaration.js]: remove ?

File: packages/composer-connector-web/lib/webconnection.js [hunk 9]
```diff
@@ -224,6 +224,9 @@ class WebConnection extends Connection {
         return businessNetwork.toArchive()
             .then((businessNetworkArchive) => {
                 const initArgs = {};
+
+                console.log('***********************************************************************');
```
Comment (davidkel) [inline packages/composer-connector-web/lib/webconnection.js]: Do we really want this console.log ?

File: packages/composer-runtime/lib/context.js [hunk 10]
```diff
@@ -758,6 +759,17 @@ class Context {
         return this.identityManager;
     }
 
+        /**
+     * Get the resource manager.
+     * @return {IdentityManager} The identity manager.
```
Comment (davidkel) [inline packages/composer-runtime/lib/context.js]: Shouldn't it be ResourceManager and The resource manager ?

File: packages/composer-runtime/lib/engine.js [hunk 11]
```diff
@@ -250,6 +254,7 @@ class Engine {
                     });
             })
             .then(() => {
+                LOG.error(method,'can you see me?');
```
Comment (davidkel) [inline packages/composer-runtime/lib/engine.js]: I can see you!!!!

File: packages/composer-runtime/lib/registrymanager.js [hunk 12]
```diff
@@ -21,17 +21,20 @@ const ParticipantDeclaration = require('composer-common').ParticipantDeclaration
...
-    'Network': 'Network'
+    'Network': 'Network',
+    'Historian':'HistorianRegistry'
 };
 const VIRTUAL_TYPES = [
     'AssetRegistry',
     'ParticipantRegistry',
     'TransactionRegistry',
-    'Network'
+    'Network','HistorianRegistry'
```
Comment (davidkel) [inline packages/composer-runtime/lib/registrymanager.js]: newline, to be consistent with the other array entries ?

File: packages/composer-runtime/lib/resolver.js [hunk 13]
```diff
@@ -349,6 +349,7 @@ class Resolver {
             // Special case for this one!
             registryId = 'default';
         } else {
+            //LOG.debug(method, 'Unsupported class declaration type ' + classDeclaration.toString());
```
Comment (davidkel) [inline packages/composer-runtime/lib/resolver.js]: keep or remove ?

File: packages/composer-client/lib/registry.js (hunk 4 above)
Comment (mbwhite) [inline packages/composer-client/lib/registry.js]: removed

File: packages/composer-client/lib/registry.js (hunk 3 above)
Comment (mbwhite) [inline packages/composer-client/lib/registry.js]: removed

File: packages/composer-client/lib/registry.js (hunk 2 above)
Comment (mbwhite) [inline packages/composer-client/lib/registry.js]: Comment added

File: packages/composer-client/lib/registry.js (hunk 5 above)
Comment (mbwhite) [inline packages/composer-client/lib/registry.js]: removed

File: packages/composer-connector-web/lib/webconnection.js (hunk 9 above)
Comment (mbwhite) [inline packages/composer-connector-web/lib/webconnection.js]: eeek!  removed

File: packages/composer-runtime/lib/engine.js (hunk 11 above)
Comment (mbwhite) [inline packages/composer-runtime/lib/engine.js]: not anymore!

**Pull request merged**